# Thesis project 2024
Denise Brunazzi, Student in Genomics at FaBit, Alma Mater Studiorum Universita di Bologna. 

## Aim
The aim of this study is to analyse the evolutionary relationship between ancient genomes of Faecalibacterium prausnitzii and their modern relatives, in order to examine the effect of industrializa9on on this specific species of bacteria present in the human gut microbiome. This study utilizes a dataset comprising of sequences of ancient metagenomes and modern assembled genomes of F. prausnitzii, collected from previous literature. Using the sequences of ancient metagenomes, metagenome-assembled genomes (MAGs) of F. prausnitzii are assembled. The ancient genomes are incorporated alongside modern genomes to construct a detailed phylogenetic tree. The constructed phylogenetic tree serves as a foundtional tool for identifying distinct clades within the genome panel, allowing for a clear visualiza9on of evolutionary relationships and divergence among the genomes. Within these identified clades, a functional analyses is conducted to investigate the conservation, gain, or loss of specific gene functions over time. 

This analysis enables us to develop hypotheses regarding the evolutionary history of F. prausnitzii, particularly focusing on how industrialization has impacted the human gut microbiome.

## Description
### `metaphlan4.sh`
This code allows to determine the microbial composition of the samples used. MetaPhlAn4 is a computational tool which identifies the microbiota species or strains present in a sample, based on specific unique genomic sequences. It generates a .tsv file containing the abundances of specific species and strains inside the sample. This files are merged together in a table containing the sequence reads in the columns, and all the identified bacterial species and subspecies in the rows. For each sequence read and bacterial species, the relative abundance is shown. If a species is unknown or not yet classified, the species name is represented as an acronym.

### `metawrap.sh`
This code is used to assemble high-quality drac genomes (metagenomic bins), refine them, and consolidate them. MetaWRAP is a modular pipeline for shotgun metagenomic data analysis which performs read quality control, assembly, bin extraction and analysis. The module `metaWRAP::Assembly` is used to assemble a set of metagenomic reads, starting from the two FASTA files resulting from the shotgun sequencing of each sample. For this study, MegaHit was used (`--megahit` flag), as it is memory efficient, faster, and scales well with large datasets. The contigs obtained are sorted by length, and their naming includes the con9g ID, length, and coverage.

The `metaWRAP::Binning` module groups con9gs into bins, where each bin represents a drac genome. It uses three different metagenomic binning socwares: MaxBin2, CONCOCT, metaBAT2 (`--maxbin2`, `--concoct`, `--metabat2` flags). For each sample, the module indexes (sorts) the metagenomic assembly obtained before, and then aligns paired end reads. The bins can be generated by one or mul9ple of the three binning socwares, and this is indicated in the output file produced.

The module `metaWRAP::Bin_refinement` compares bin sets to each other iteratively, producing consolidated, improved bin sets. After all bin sets have been consolidated, a de-replication function removes any duplicate con9gs.

Bins were selected if they have >50% completeness and <5% contamina9on (`-c 50`, `-x 5` flags).

### `classification_and_analysis.sh`
GTDB-Tk is used to identify those MAGs containing F. prausnitzii.

The average nucleotide identity (ANI) pairwise distances for the whole genomes in the genome panel were computed using `pyani`, a python-based program. The alignment socware used was BLAST (`-m ANIb` flag). The process produces a _.tab_ file or the alignment lengths and another _.tab_ file for the percentage identities.

Each genome of the genome panel was analysed using Prokka, a software tool to rapidly annotate de novo assembled bins and identify coding sequences (ORFs) in prokaryotic genomes. Prokka saves the ORF to a _FASTA_ file, translates it into an amino acid sequence, aligns the sequence to a database of annotated sequences, and determines the function of the sequence.

The _.gff3_ files produced by Prokka, containing the nucleotide sequences, are used by ROARY, a pipeline that generates a pangenome. For this study, the pipeline was set to generate clusters with â‰¥90% percentage iden9ty, and to define as core genomes those present in â‰¥90% of genomes (`-i 90`, `-cd 90` flags). The process produces a gene presence/absence table.

### `phylogenetic_tree.sh`
The tree was built using PhyloPhlAn, an integrated pipeline for large-scale phylogenetic profiling of genomes and metagenome. It assigns MAGs to species-level genome bins (SGBs), and it reconstructs phylogenies using clade-specific phylogenetic markers. The configuration file was customized using BLAST Diamond for the mapping step (`--db_aa diamond`, `--map_dna diamond`, `--map_aa diamond`), MAFFT for the mul9ple sequence alignment (`--msa mafft`), trimAl for trimming (`--trim trimal`), FastTree to build the first tree (`--tree1 fastree`), and RAxML to refine the final tree (`--tree2 raxml`). The additional parameters used for the phylogene9c analysis were `--diversity low` and `--fast`. 
















